### ISSUE TITLE
Using @Schema disables enum value generation

### ISSUE THREAD

**Issue by neon-dev** [smallrye/smallrye-open-api#110]
Annotating an enum field with for example `@Schema(required = true)` stops auto generation of enum value info.
This unfotunately cannot be restored by specifying `@Schema(implementation = MyEnum.class, required = true)`.
There should be a way. Best would be to not having to specify an implementation at all. I think the implementation should in fact be only overriden if you explicitly specify one. If not, the default generation should be used.
My proposal is to change `Class<?> implementation() default Void.class;` to `Class<?> implementation() default Infer.class;`, where `Infer.class` is just an internal control class. You will then have to specify `Void.class` yourself to disable auto generation.

**Comment by EricWittmann** [comment]
It's interesting - we're going to be talking about default annotation property values in next week's Microprofile OpenAPI community call.  My current understanding of things is that the default values for all MP+OAI annotations are intended to indicate "not specified" and therefore what SmallRye **should** be doing in this particular use-case is to use the implementation class in exactly the way you are expecting.  It should not actually use Void.

So the upshot of this is that I suspect we're simply looking at a bug in the impl here.

**Comment by neon-dev** [comment]
Yes, I have to admit I've stumpled upon counter-intuitive behavior multiple times already.
For example using `@RequestBody(required = true)` currently behaves the same. Therefore you have to write `@RequestBody(name = "tokenRequestBody", content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = MyRequest.class)), required = true)` which is not convenient at all.

Thanks for looking into it. Would be great not having to declare redundant info unless really necessary.

**Comment by neon-dev** [comment]
By the way, since we're talking about default properties and I was just mentioning `@RequestBody`. It should be considered making required = true the default here, because more often than not you expect and depend on it.
This would make writing `@RequestBody(required = true)` obsolete, which I find a good thing. But that's just my take on it.

**Comment by EricWittmann** [comment]
Any chance you would be willing to raise that as an issue in the spec?  

https://github.com/eclipse/microprofile-open-api/issues

It may not be something the spec will need to address (e.g. it could be vendor specific) but it would be nice to have that conversation in the spec.

**Comment by neon-dev** [comment]
Done. Have a nice weekend 😄

**Comment by neon-dev** [comment]
This is somewhat related, so I wanted to ask before creating another issue:
Would it be reasonable to suggest making method parameters annotated with `@QueryParam` **not** required by default? Don't know if it's implementation specific or there's a spec that defines this.
I feel like in contrast to `@PathParam` or `@FormParam` they are often less essential.

**Comment by EricWittmann** [comment]
That's an interesting proposition.  Definitely please raise it as a spec issue and we'll discuss in the next spec hangout.

**Comment by neon-dev** [comment]
Was a bit unconfident but I hope I created the issue correctly. If not, let me know what to change @EricWittmann.

**Comment by EricWittmann** [comment]
It looks good to me - I've tagged it appropriately.

**Comment by MikeEdgar** [comment]
I took a look and I see why the original issue regarding missing `Enum` values is occurring. I ran in to some strange behavior with enums myself so fixing this would be a help. I can do a PR if no one else is looking at it yet. Is the correct result that enumerated values in the `@Schema` annotation should override those in the `Enum` typed class, or are they meant to be merged with them? To me, the more obvious answer would be that they override them. Thoughts?

There was also a question in this thread (@neon-dev) regarding non-required params. I think the behavior you're seeing is fixed, but just not yet released. It looks like the OpenAPI spec does define the default requirements for parameters, which aligns with what you were thinking: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#parameterObject
